src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -310,6 +310,145 @@ def k8s_install_cli(client_version='latest', install_location=None):
+        image_tag = 'canary'
+    # Official chart URL to install
+
+    logger.warning('Deploying the aci-connector using Helm')
+    try:
+        subprocess.call(["helm", "install", chart_url, "--name", name, "--set", "env.azureClientId=" +
+                         service_principal + ",env.azureClientKey=" + client_secret + ",env.azureSubscriptionId=" +
+                         subscription_id + ",env.azureTenantId=" + tenant_id + ",env.aciResourceGroup=" + rgkaci.name +
+                         ",env.aciRegion=" + location + ",image.tag=" + image_tag])
+    except subprocess.CalledProcessError as err:
+        raise CLIError('Could not deploy the ACI Chart: {}'.format(err))
+
+def k8s_uninstall_connector(client, name, name_aks, resource_group_name_aks,
+                            graceful=False, linux=True, windows=False):
+    """Undeploy the ACI-Connector from an AKS cluster.
+    :param name_aks: The name of the AKS cluster. The name is case insensitive.
+    :type name_aks: str
+    :param resource_group_name_aks: The name of the resource group where the AKS cluster is deployed.
+    :type resource_group_name_aks: str
+    :param name: The name for the ACI Connector
+    :type name: str
+    :param graceful: Mention if you want to drain/uncordon your aci-connector to move your applications
+    running on ACI to your others nodes. Default : False
+    :type graceful: bool
+    :param windows: Os type target will be Windows for the connector
+    :type windows: bool
+    :param linux: Os type target will be Linux for the connector
+    :type linux: bool
+    """
+    from subprocess import PIPE, Popen
+    helm_not_installed = "Error : Helm not detected, please verify if it is installed."
+    # Check if Helm is installed locally
+    try:
+        p = Popen(["helm"], stdout=PIPE, stderr=PIPE)
+    except OSError:
+        raise CLIError(helm_not_installed)
+    # Get the credentials from a AKS instance
+    _, browse_path = tempfile.mkstemp()
+    aks_get_credentials(client, resource_group_name_aks, name_aks, admin=False, path=browse_path)
+    # Validate if the RG exists
+    # Just do the get, we don't need the result, it will error out if the group doesn't exist.
+    if graceful:
+        logger.warning('Graceful option selected, will try to drain the node first')
+        kubectl_not_installed = "Error : Kubectl not detected, please verify if it is installed."
+        try:
+            p = Popen(["kubectl"], stdout=PIPE, stderr=PIPE)
+        except OSError:
+            raise CLIError(kubectl_not_installed)
+        try:
+            if windows:
+                drain_node = subprocess.check_output(
+                    ["kubectl", "drain", "aci-connector-0", "--force"],
+                    universal_newlines=True)
+                drain_node = subprocess.check_output(
+                    ["kubectl", "drain", "aci-connector-1", "--force"],
```

Comment:
same here. show we use {name}-win, {name}-linux as the node name?

Reply:
This is not supported in the current connector codebase, we cannot specify a custom node name

Reply:
see @itowlson last comment